addons/payment_authorize/models/payment_transaction.py

```diff
@@ -95,18 +95,36 @@ def _send_refund_request(self, amount_to_refund=None, create_refund_transaction=
                 create_refund_transaction=create_refund_transaction,
             )
 
+        authorize_API = AuthorizeAPI(self.acquirer_id)
+        tx_details = authorize_API.get_transaction_details(self.acquirer_reference)
+        create_refund_transaction = tx_details and (not tx_details.get('err_code'))
+
         refund_tx = super()._send_refund_request(
-            amount_to_refund=amount_to_refund, create_refund_transaction=False
+            amount_to_refund=self.amount, create_refund_transaction=create_refund_transaction
         )
 
-        authorize_API = AuthorizeAPI(self.acquirer_id)
-        rounded_amount = round(self.amount, self.currency_id.decimal_places)
-        res_content = authorize_API.refund(self.acquirer_reference, rounded_amount)
+        if tx_details and tx_details.get('err_code'):
+            # In the case of an error a 'res_content' must be passed back to the
+            # '_handle_feedback_data' anyways.
+            res_content = {
+                'x_response_code': AuthorizeAPI.AUTH_ERROR_STATUS,
+                'x_response_reason_text': tx_details.get('err_msg')
+            }
+        elif (tx_details.get('transaction', {}).get('transactionStatus')
+              in ['capturedPendingSettlement', 'authorizedPendingCapture']):
+            # For Authorize captured transactions are not automotically debited.
+            # Transactions with status in 'captured, pending settlement' and 'authorised,
+            # pending capture' are voided if a refund is asked, as the credit was not moved
+            # yet.
+            res_content = authorize_API.void(self.acquirer_reference)
+        else:
+            # continue with the usual refund
+            # We continue to use the full amount below
+            rounded_amount = round(amount_to_refund, self.currency_id.decimal_places)
```

Comment:
To be consistent, you may want to choose between `amount_to_refund` (which will be the full amount due to the fact it's `full_only`), or use `self.amount` as in line 103. 
Also, is the comment needed here? If you feel it's, then don't mind me. :)